R/froll.R

```diff
@@ -111,6 +111,47 @@ make.roll.names = function(x.len, n.len, n, x.nm, n.nm, fun, adaptive) {
   ans
 }
 
+# irregularly spaced time series, helper for creating adaptive window size
+frolladapt = function(x, n, align="right", partial=FALSE, give.names=FALSE) {
+  x = unclass(x)
+  if (!is.numeric(x))
+    stopf("'x' must be of a numeric type")
+  if (!is.integer(x))
+    x = as.integer(x)
+  if (!is.numeric(n)) {
+    stopf("'n' must be an integer")
+  } else {
+    nms = names(n) ## only for give.names
+    if (!is.integer(n)) {
+      if (!fitsInInt32(n))
+        stopf("'n' must be an integer")
+      n = as.integer(n)
+    }
+  }
+  if (!length(n))
+    stopf("'n' must be non 0 length")
+  if (anyNA(n))
+    stopf("'n' must not have NAs")
+  if (!identical(align, "right"))
+    stopf("'align' other than 'right' has not yet been implemented")
+  if (!isTRUEorFALSE(partial))
+    stopf("'%s' must be TRUE or FALSE", "partial")
+  if (!isTRUEorFALSE(give.names))
+    stopf("'%s' must be TRUE or FALSE", "give.names")
+
+  if (length(n) == 1L) {
+    ans = .Call(Cfrolladapt, x, n, partial)
+  } else {
+    ans = lapply(n, function(.n) .Call(Cfrolladapt, x, .n, partial))
```

Comment:
I can imagine it's more efficient to iterate over each `x` for a large number of `n`, rather than recompute the full window at each iteration, right?

just checking my understanding. we can revisit that much later if it becomes a highly-requested use case.

Reply:
`n` a vector means to computer multiple window sizes, I believe usually will be 1, then sometimes 2, maybe 4. In rather unlikely cases more.
If we would want to loop over `x` elements (which is the data observations itself, so much longer than `n`), then we would have to use different implementation of C routine.
If really big `n` vector would eventually being highly requested, I would rather prefer to switch lapply to mclapply than have two C implementations.